src/linear_algebra/finsupp.lean

```diff
@@ -589,6 +589,45 @@ def lcongr {ι κ : Sort*} (e₁ : ι ≃ κ) (e₂ : M ≃ₗ[R] N) : (ι →
   (i : ι) (m : M) : lcongr e₁ e₂ (finsupp.single i m) = finsupp.single (e₁ i) (e₂ m) :=
 by simp [lcongr]
 
+section sum
+
+variables (R)
+
+/-- The linear equivalence between `(α ⊕ β) →₀ M` and `(α →₀ M) × (β →₀ M)`. -/
+@[simps apply symm_apply] def sum_arrow_lequiv_prod_arrow {α β : Type*} :
+  ((α ⊕ β) →₀ M) ≃ₗ[R] (α →₀ M) × (β →₀ M) :=
+{ map_add' :=
+    by { intros, ext;
+          simp only [equiv.to_fun_as_coe, prod.fst_add, prod.snd_add, add_apply,
+              snd_sum_arrow_equiv_prod_arrow, fst_sum_arrow_equiv_prod_arrow] },
```

Comment:
I think it's probably worth adding a `((α ⊕ β) →₀ M) ≃+ (α →₀ M) × (β →₀ M)` version of this in `finsupp/basic` like we do for `map_domain` and `map_range`, and moving this proof there.

Reply:
I added `finsupp.sum_arrow_add_equiv_prod_arrow`. What do you think about renaming them to `{,linear_,add_}equiv.sum_finsupp_equiv_prod_finsupp`?

Reply:
I like replacing `arrow` with `finsupp`, but I don't think the name spacing is useful. Note that this actually works with dot notation where you have it right now. So maybe `finsupp.sum_add_equiv_prod`?

Reply:
It's unfortunate that `prod` and `sum` are overloaded...